mapie/classification.py

```diff
@@ -366,8 +373,14 @@ def predict(
                     interpolation="lower"
                 ) for _alpha in alpha_
             ])
-            prediction_sets = np.stack([
-                y_pred_proba > (quantile)
-                for quantile in self.quantiles_
-            ], axis=2)
+            if self.method == "score":
+                prediction_sets = np.stack([
+                    y_pred_proba > (quantile)
+                    for quantile in self.quantiles_
+                ], axis=2)
+            else:
+                prediction_sets = np.stack([
```

Comment:
Pareil ici, il faut inclure les classes avec les probas les plus élevées afin que leur cumul dépasse le quantile calculé. Je vois ça en plusieurs étapes:
- sort probabilities by descending order
- get corresponding classes
- compute number of classes to include in prediction sets
- get corresponding classes
- convert to boolean table

Je propose le code ci-dessous, mais qui est un peu lourd et pas très lisible. Peux-tu essayer de l'améliorer stp ?

```
                # sort probabilities by descending order
                y_pred_sorted = -np.sort(-y_pred_proba)
                # get corresponding order of classes
                y_pred_sorted_classes = np.argsort(-y_pred_proba)
                # compute number of classes to include in prediction sets
                num_classes_sorted = np.stack([
                    np.cumsum(y_pred_sorted, axis=1) <= (quantile)
                    for quantile in self.quantiles_
                ], axis=2).sum(axis=1) + 1
                # get corresponding classes
                prediction_sets_int = np.array([
                    [
                        pred[:num_classes_sorted[j, i]]
                        for i, _ in enumerate(self.quantiles_)
                    ] for j, pred in enumerate(y_pred_sorted_classes)
                ])
                # convert to boolean table
                y_pred_labels = np.tile(np.arange(3), (y_pred.shape[0], 1))
                prediction_sets = np.stack([
                    [
                        np.in1d(pred_labels, prediction_sets_int[i, j])
                        for i, pred_labels in enumerate(y_pred_labels)
                    ] for j, _ in enumerate(self.quantiles_)
                ], axis=2)
```

Reply:
j'ai gardé la même solution pour le moment

Reply:
Nouvelle implémentation mieux optimisée avec un test unitaire associé.